src/core/server/elasticsearch/elasticsearch_config.ts

```diff
@@ -65,6 +65,7 @@ export const config = {
     }),
     apiVersion: schema.string({ defaultValue: DEFAULT_API_VERSION }),
     healthCheck: schema.object({ delay: schema.duration({ defaultValue: 2500 }) }),
+    ignoreVersionMismatch: schema.boolean({ defaultValue: false }),
```

Comment:
Oddly, I'm now getting "Unknown config key: elasticsearch.ignoreVersionMismatch", even though the config is defined here. Or am I missing something?

Reply:
Huh, seems like I need to `server.config().has('elasticsearch.ignoreVersionMismatch')` before I can read the config. Anybody knows why that is necessary when a default value (`false`) is defined?

Reply:
Okay, that didn't fix it `server.config().has('elasticsearch.ignoreVersionMismatch')` returns `false` regardless if it's set in kibana.yml or not. Seems like I cannot access the elasticsearch config values from here. Any pointers appreciated 🙏

Reply:
Looks like you still need to expose the `ignoreVersionMismatch` as a public property on the ElasticSearchConfig class in this same file. We wrap the config in a class before exposing it.

Reply:
Okay, turns out I cannot access the elasticsearch config from here:
https://github.com/elastic/kibana/blob/b6591eb0bdde65c03d144ce1c9483e477f8eb96d/src/core/server/legacy/legacy_service.ts#L47

Got it working by passing the config value down through. Thanks @rudolf !